src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+                sectionPanel.setSectionNameForDisplay(sectionName.equals(Const.DEFAULT_SECTION) ? DISPLAY_NAME_FOR_DEFAULT_SECTION 
+                                                                                                : sectionName);
+                sectionPanel.setDisplayingTeamStatistics(true);
+                break;
+            default:
+                Assumption.fail();
+                break;
+        }
+    }
+    
+
+    private void buildMissingTeamAndParticipantPanelsWithoutModerationButtonForSection(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String section,
+                                    Set<String> receivingTeams) {
+        buildMissingTeamAndParticipantPanelsForSection(sectionPanel, section, receivingTeams, 
+                                                       false);
+    }
+    
+    private void buildMissingTeamAndParticipantPanelsWithModerationButtonForSection(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String section,
+                                    Set<String> receivingTeams) {
+        buildMissingTeamAndParticipantPanelsForSection(sectionPanel, section, receivingTeams, 
+                                                       true);
+    }
+    
+    private void buildMissingTeamAndParticipantPanelsForSection(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String sectionName,
+                                    Set<String> receivingTeams,  
+                                    boolean isWithModerationButton) {
+
+        // update the teams for the previous section
+        Set<String> teamsInSection = bundle.getTeamsInSectionFromRoster(sectionName);
+        Set<String> teamsWithoutResponses = new HashSet<String>(teamsInSection);
+        teamsWithoutResponses.removeAll(receivingTeams);
+        
+        // create for every remaining team in the section, participantResultsPanels for every team member
+        for (String teamWithoutResponses : teamsWithoutResponses) {
+            List<String> teamMembersOfTeam = new ArrayList<String>(bundle.getTeamMembersFromRoster(teamWithoutResponses));
+            Collections.sort(teamMembersOfTeam);
+            if (isWithModerationButton) {
+                addMissingParticipantsForTeamToSectionPanelWithModerationButton(sectionPanel, teamWithoutResponses, teamMembersOfTeam);
+            } else {
+                addMissingParticipantsForTeamToSectionPanelWithoutModerationButton(sectionPanel, teamWithoutResponses, teamMembersOfTeam);
+            }
+        }
+        
+    }
+
+    private String getCurrentTeam(FeedbackSessionResultsBundle bundle, String giverIdentifier) {
+        String currentTeam;
+        if (bundle.isParticipantIdentifierInstructor(giverIdentifier)) {
+            currentTeam = Const.USER_TEAM_FOR_INSTRUCTOR;
+        } else {
+            currentTeam = bundle.getTeamNameForEmail(giverIdentifier);
```

Comment:
I think I will leave this here for now, improving on this and fixing the buggy methods in the bundle for obtaining team name should be done in that other issue